bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusThermostatHandler.java

```diff
@@ -0,0 +1,265 @@
+                            + thermostatId);
+            return;
+        }
+
+        Map<Integer, QbusThermostat> thermostatComm = qComm.getThermostat();
+
+        if (thermostatComm != null) {
+            QbusThermostat qThermostat = thermostatComm.get(thermostatId);
+
+            if (qThermostat == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Bridge communication not initialized when trying to execute command for Scene "
+                                + thermostatId);
+                return;
+            } else {
+                scheduler.submit(() -> {
+                    if (!qComm.communicationActive()) {
+                        restartCommunication(qComm, "Thermostat", thermostatId);
+                    }
+
+                    if (qComm.communicationActive()) {
+                        if (command == REFRESH) {
+                            handleStateUpdate(qThermostat);
+                            return;
+                        }
+
+                        switch (channelUID.getId()) {
+                            case CHANNEL_MODE:
+                                handleModeCommand(qThermostat, command);
+                                break;
+
+                            case CHANNEL_SETPOINT:
+                                handleSetpointCommand(qThermostat, command);
+                                break;
+
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Puts thing offline
+     *
+     * @param message
+     */
+    public void thingOffline(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, message);
+    }
+
+    /**
+     * Executes the Mode command
+     */
+    private void handleModeCommand(QbusThermostat qThermostat, Command command) {
+        String snr = getSN();
+        if (command instanceof DecimalType) {
+            int mode = ((DecimalType) command).intValue();
+            if (snr != null) {
+                qThermostat.executeMode(mode, snr);
+            } else {
+                thingOffline("No serial number configured for  " + thermostatId);
+            }
+        }
+    }
+
+    /**
+     * Executes the Setpoint command
+     */
+    private void handleSetpointCommand(QbusThermostat qThermostat, Command command) {
+        String snr = getSN();
+        if (command instanceof QuantityType<?>) {
+            QuantityType<?> s = (QuantityType<?>) command;
+            double sp = s.doubleValue();
+            QuantityType<?> spCelcius = s.toUnit(CELSIUS);
+
+            if (snr != null) {
+                if (spCelcius != null) {
+                    qThermostat.executeSetpoint(sp, snr);
+                } else {
+                    thingOffline("Could not set setpoint for thermostat (convertion failed)  " + thermostatId);
```

Comment:
As this is not a permanent error, you are good to log it only.
```suggestion
                    thingOffline("Could not set setpoint for thermostat (conversion failed)  " + thermostatId);
```